frontend/src/components/Events/EventList.jsx

```
@@ -155,54 +188,30 @@ const EventList = () => {
-                    }
+                    onClick={() => handleRSVP(event.id, "interested", isInterested)}
                     disabled={actionLoading[event.id]}
-                    className={`w-full flex items-center justify-center py-1 px-3 bg-blue-500 text-white rounded-lg transition-all duration-300 text-xs ${
-                      actionLoading[event.id]
-                        ? "opacity-50 cursor-not-allowed"
-                        : isInterested
-                        ? "hover:bg-blue-700"
-                        : "hover:bg-blue-600"
-                    }`}
+                    className={`w-full flex items-center justify-center py-1 px-3 bg-blue-500 text-white rounded-lg transition-all duration-300 text-xs ${interestedButtonClass}`}
                   >
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)